### PR TITLE
spades: build with cmake

### PR DESCRIPTION
**Pull request by hubutui**
## Involved packages

 - spades

## Details

- [x] Tested in the local machine (largest 16G RAM) and it is passed without any issue
- [ ] Provide New Package
- [x] Fix the Packages
  - [x] PKGBUILD
  - [ ] lilac.yaml
  - [ ] lilac.py
- [ ] Would like to continue to work with us

### CODE REVIEW

**Comment by hubutui** [comment]
@alienzj Hello, would you like to review this pull request? BTW, according to the latest [build log](https://build.bioarchlinux.org/api/pkg/spades/log/1658750630). This pkg did not build with `lto` enabled, you might wanna report this issue to upstream.

File: BioArchLinux/spades/PKGBUILD [hunk 1]
```diff
@@ -9,84 +7,34 @@ pkgrel=1
...
-    cd "$srcdir/$pkgname-$pkgver/assembler" || exit
-
-    ./spades_compile.sh
-}
-
-check() {
-    cd "$srcdir/$pkgname-$pkgver/assembler/bin/" || exit
-
-    python spades.py --test
+  cmake -B ${pkgname}-${pkgver}/assembler/build_spades -S ${pkgname}-${pkgver}/assembler/src \
```
Comment (alienzj) [inline BioArchLinux/spades/PKGBUILD:28]: Great!

File: BioArchLinux/spades/PKGBUILD [hunk 2]
```diff
@@ -9,84 +7,34 @@ pkgrel=1
...
-    'bzip2'
-    'bash'
-    'openmp'
+  bzip2
+  cmake
+  openmp
+  zlib
 )
-checkdepends=('python')
+options=("!lto")
```
Comment (alienzj) [inline BioArchLinux/spades/PKGBUILD:20]: I don't know this part. Why it influences the build process ?

Comment (alienzj) [reply BioArchLinux/spades/PKGBUILD:20]: Report it to upstream:
https://github.com/ablab/spades/issues/1007

Comment (alienzj) [reply BioArchLinux/spades/PKGBUILD:20]: The author suggested us just do not build with LTO.
I think we can merge this PR.

Comment (starsareintherose) [reply BioArchLinux/spades/PKGBUILD:20]: I think you can merge by yourself

Comment (alienzj) [reply BioArchLinux/spades/PKGBUILD:20]: OK.

File: BioArchLinux/spades/PKGBUILD (hunk 1 above)
Comment (hubutui) [reply BioArchLinux/spades/PKGBUILD:28]: I should change `${pkgname}-${pkgver}/assembler/build_spades` to `build`. The `python spades --test` only works if we use the default `PREFIX`.

**Pull request merged**